File: src/light.cpp
#include "light.h"
#include "tuple.h"
#include "color.h"
#include "comparinator.h"

PointSource::PointSource()
{
    this->mbrPosition = Point(0,0,0);
    this->mbrIntensity = Color(0,0,0);
}

PointSource::PointSource(Point position, Color intensity)
{
    this->mbrPosition = position;
    this->mbrIntensity = intensity;
}

bool PointSource::checkEqual(const PointSource &other) const
{
    Comparinator ce = Comparinator();
    return ce.checkTuple(this->mbrPosition,other.mbrPosition) && mbrIntensity.checkEqual(other.mbrIntensity);
}

File: src/Light.cs
using LibColor;
using LibTuple;
using LibComparinator;
namespace LibLight;

public class PointSource {
    public Point mbrPosition;
    public Color mbrIntensity;
    private Comparinator _fieldComp = new Comparinator();
    public PointSource()
    {
        mbrPosition = new Point(0,0,0);
        mbrIntensity = new Color(1,1,1);
    }
    public PointSource(Point position, Color intensity)
    {
        mbrPosition = position;
        mbrIntensity = intensity;
    }
    public bool CheckEqual(PointSource other)
    {
        return _fieldComp.CheckTuple(mbrPosition,other.mbrPosition) && mbrIntensity.CheckEqual(other.mbrIntensity);
    }
};